articles/why-and-how-zk-snark-works/3.Non_Interactive_Zero-Knowledge_of_a_Polynomial.md

```diff
@@ -0,0 +1,162 @@
+
+$$h(x) = \frac{p(x)}{t(x)}$$
+
+In particular:
+Given:
+$$ p(x)  = x^3 - 3x^2 + 2x $$ 
+$$ t(x) = (x-1)(x-2) $$
+
+The Verifier samples a random value 23, calculates  $$t = t(23) = (23 - 1)(23 - 2) = 462$$ and gives 23 to the Prover.
+
+The Prover calculates $h(x) = p(x)$, $t(x) = x$, evaluates $$ p = p(23) = 10626 $$ $$h = h(23) = 23 $$ and provides p, h to the Verifier.
+
+The Verifier then checks that:  $$p = t \cdot h: 10626 = 462 \cdot 23$$ 
+This is true, and therefore the statement is proven.
+
+## 3.2. Modular Arithmetic 
+
+## 3.3. Encrypted Polynomial
+
+Let's define *E* : $$E(x) = g^{v} \pmod{n}$$
+
+For example, we have:
+$$ p(x) = x^{3} - 3x^{2} + 2x$$
+
+In this case, with coefficients 1, -3, 2, the encrypted polynomial can be evaluated as follows:
+
+$$
+\begin{aligned}
+E(x^{3})^{1}E(x^{2})^{-3}E(x)^{2}= \\
+ (g^{x^{3}})^{1}(g^{x^{2}})^{-3}(g^{x})^{2}= \\
+ g^{1x^{3}-3x^{x^2}g^{2x}}= \\
+ g^{x^{3}-3x^{2}+2x} \\
+\end{aligned}
+$$
+
+*Verifier*:
+
+* Generate a random *s*
+* Calculate $E(s^{i}) = g^{s^{i}}$: same as the example above
+* Evaluate the *target polynomial* with *s*: *t(s)*
+* Provide to the Prover: $E(s^{0}),E(s^{1}),E(s^{2}),...,E(s^{d})$
+
+*Prover*:
+
+* Calculates $h(x) = \frac{p(x)}{t(x)}$
+* Evaluates: $E(p(s))$, $E(h(s))=g^{h(s)}$
+* Provides $g^{p}$, $g^{h}$
+
+*Verifier*:
+
+* Check that $p=t(s) \cdot h$, meaning: 
+  $g^{p} = (g^{h})^{t(s)}$
+
+If the Prover claims to have a satisfactory polynomial using only 2 powers $s^{3}$ and $s^{1}$, it is not possible to verify in the current protocol.
+
+## 3.4. Restricting a Polynomial
+
+The main idea is to sample a random $\alpha$, and the provider needs to calculate $a^{,} = a^{\alpha} \pmod{n}$, then provide the tuple $(a,a^{,})$ using an elliptic curve. This ensures that the provider can't extract $\alpha$ from the tuple $(a,a^{,})$ through brute force.
+
+We can also apply this construction with the simple one-coefficient polynomial $f(x)= c \cdot x$.
+
+The Verifier can now check the equality:  $(a^{c})^{\alpha} = (\alpha^{,})^{c}$
+
+In summary:
+
+*Verifier*:
+
+* Chooses 2 random *s*, $\alpha$ and provides evaluations for $x=s$ and $(g^{s},g^{\alpha \cdot s})$
+* *Prover*: Applies the coefficient c:  $((g^{s})^c,(g^{\alpha \cdot s})^{c})=(g^{c \cdot s},g^{\alpha \cdot s \cdot c})$
+* *Verifier*: Checks
+
+In particular, for a degree $d$ polynomial: 
+$$p(x) = x^{3} - 3x^{2} + 2x$$ 
+
+*Verifier*:
+
+* Provides encrypted: $g^{s^{0}},g^{s^{1}}, \cdots,g^{s^{d}}$ and their *shifts* $g^{\alpha s^{0}},g^{\alpha s^{1}}, \cdots,g^{\alpha s^{d}}$
+
+*Prover*:
+
+* Evaluates the encrypted polynomial with $s$: 
+$$g^{p} = g^{p(s)} =(g^{s_{0}})^{c_{0}} \cdot (g^{s_{1}})^{c_{1}} \cdots (g^{s_{d}})^{c_{d}} = g^{c_{0}s^{0} +c_{1}s^{1}+ \cdots+c_{d}s^{d}  } =g^{s^{3}} \cdot g^{-3s^{2}} \cdot g^{2s}=g^{s^{3}-3s^{2}+2s}$$       
+$$\Rightarrow g^{{p}'} = g^{\alpha (s^{3}-3s^{2}+2s)}$$ 
+*Verifier* checks :
```

Comment:
Formatting error and also, can you provide more information on this? I can barely understand this part.